packages/editor/src/components/post-locked-modal/index.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * External dependencies
+ */
+import jQuery from 'jquery';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Modal, Button } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import { Component } from '@wordpress/element';
+import { compose, withGlobalEvents } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { getWPAdminURL } from '../../utils/url';
+import PostPreviewButton from '../post-preview-button';
+
+class PostLockedModal extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.sendPostLock = this.sendPostLock.bind( this );
+		this.receivePostLock = this.receivePostLock.bind( this );
+		this.releasePostLock = this.releasePostLock.bind( this );
+	}
+
+	componentDidMount() {
+		// Details on these events on the Heartbeat API docs
+		// https://developer.wordpress.org/plugins/javascript/heartbeat-api/
+		jQuery( document )
+			.on( 'heartbeat-send.refresh-lock', this.sendPostLock )
+			.on( 'heartbeat-tick.refresh-lock', this.receivePostLock );
```

Comment:
@adamsilverstein is there a reason we need to rely on `jQuery` here? could we have gotten by with `addEventListener()` (and for IE `attachEvent`)?

given the assumption that WordPress is already loaded `jQuery` seems free, but for external uses these few lines can pull in a huge dependency

thoughts?

Reply:
We should do similar as to what was done in #8311 as a short-term solution.

Reply:
In fact, at least for the `heartbeat-tick` event, we should already have the action provided through #8311.

Reply:
Pull request at #11781